src/dshould/ShouldType.d

```diff
@@ -0,0 +1,218 @@
+module dshould.ShouldType;
+
+import std.algorithm : map;
+import std.format : format;
+import std.range : iota;
+import std.string : empty, join;
+import std.traits : TemplateArgsOf;
+public import std.traits : isInstanceOf;
+import std.typecons : Tuple;
+
+public auto should(T)(lazy T got) pure
+{
+    T get() pure @safe
+    {
+        return got;
+    }
+
+    return ShouldType!(typeof(&get))(&get);
+}
+
+public struct ShouldType(G, string[] words = [])
```

Comment:
`phrase` sounds like a sequence of words, while `words` sounds like a set of words
(as in `allowedWords`)